web/assets/js/components/gameboard.ts

```diff
@@ -269,23 +286,27 @@ const Gameboard = class extends Component {
 
       // Fetch the move history to check for a capture
       const moveHistory = this.chess.history({ verbose: true });
-      if (moveHistory.length == 0) {
-        await this.engine();
+      console.log(moveHistory);
 
-        return;
+      //TODO: not sure we need it
+      if (moveHistory.length == 0) {
+        console.log(moveHistory.length == 0);
+        // Do we need this? there is a this.engine(); at the end of the function
+        //   await this.engine();
+        // return;
```

Comment:
You can cause a panic without this return, since if `moveHistory.length === 0`, then L299 errors out